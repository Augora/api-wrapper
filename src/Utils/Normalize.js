const { normalize } = require('normalizr');
const { depute } = require('./Schema/DeputesSchema');

function normalizeDeputes(normalizeFunction, deputesFromAPI) {
  return normalizeFunction(deputesFromAPI);
}

function normalizeDeputesFactory(normalizeFunction) {
  return normalizeDeputes.bind(undefined, normalizeFunction);
}

const providedNormalizeDeputes = normalizeDeputesFactory(data => normalize(data, [depute]));

module.exports = {
  normalizeDeputes,
  normalizeDeputesFactory,
  providedNormalizeDeputes,
};
