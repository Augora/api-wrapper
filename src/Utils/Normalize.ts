import { normalize } from "normalizr";
import { deputy } from "./Schema/DeputySchema";

export interface INormalize {
  (o: object): object;
}

const deputiesInOfficeNormalizer = (data: object) => normalize(data, [deputy]);

export function normalizeDeputes(
  normalizeFunction: INormalize,
  deputesFromAPI: object
) {
  return normalizeFunction(deputesFromAPI);
}

export function normalizeDeputesFactory(normalizeFunction: INormalize) {
  return normalizeDeputes.bind(undefined, normalizeFunction);
}

export const providedNormalizeDeputes = normalizeDeputesFactory(
  (data: object) => normalize(data, [deputy])
);

export function providedNormalizeDeputy(data: object) {
  return normalize(data, [deputy]);
}
export const providedNormalizeDeputesInOffice = normalizeDeputesFactory(
  deputiesInOfficeNormalizer
);
