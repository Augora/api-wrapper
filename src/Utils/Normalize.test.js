import { providedNormalizeDeputes } from './Normalize';
import deputesSeeds from './DeputesSeeds';

test('test normalize with custom data', () => {
  const data = [
    {
      id: 1,
      sites_web: [
        { site: 'a.fr' },
        { site: 'b.fr' },
      ],
    },
    {
      id: 2,
      sites_web: [
        { site: 'c.fr' },
        { site: 'd.fr' },
      ],
    },
  ];
  const normalizedData = providedNormalizeDeputes(data);
  expect(normalizedData).toMatchSnapshot();
});

test('test normalize with custom data', () => {
  const normalizedData = providedNormalizeDeputes(deputesSeeds);
  expect(normalizedData).toMatchSnapshot();
});
