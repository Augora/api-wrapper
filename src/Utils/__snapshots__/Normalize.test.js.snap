// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test normalize with custom data 1`] = `
Object {
  "entities": Object {
    "depute": Object {
      "1": Object {
        "id": 1,
        "sites_web": Array [
          "a.fr",
          "b.fr",
        ],
      },
      "2": Object {
        "id": 2,
        "sites_web": Array [
          "c.fr",
          "d.fr",
        ],
      },
    },
    "site": Object {
      "a.fr": Object {
        "site": "a.fr",
      },
      "b.fr": Object {
        "site": "b.fr",
      },
      "c.fr": Object {
        "site": "c.fr",
      },
      "d.fr": Object {
        "site": "d.fr",
      },
    },
  },
  "result": Array [
    1,
    2,
  ],
}
`;

exports[`test normalize with custom data 2`] = `
Object {
  "entities": Object {
    "depute": Object {
      "1": Object {
        "date_naissance": "1956-02-02",
        "groupe_sigle": "LR",
        "id": 1,
        "id_an": "230329",
        "lieu_naissance": "Paris (Paris)",
        "mandat_debut": "2012-06-20",
        "nb_mandats": 2,
        "nom": "Jean-François Lamour",
        "nom_circo": "Paris",
        "nom_de_famille": "Lamour",
        "num_circo": 13,
        "num_deptmt": "75",
        "parti_ratt_financier": "Les Républicains",
        "place_en_hemicycle": "80",
        "prenom": "Jean-François",
        "profession": "Kinésithérapeute",
        "sexe": "H",
        "sites_web": Array [
          "http://www.jflamour.fr",
          "https://twitter.com/JF_Lamour",
        ],
        "slug": "jean-francois-lamour",
        "twitter": "JF_Lamour",
        "url_an": "http://www2.assemblee-nationale.fr/deputes/fiche/OMC_PA230329",
        "url_nosdeputes": "http://www.nosdeputes.fr/jean-francois-lamour",
        "url_nosdeputes_api": "http://www.nosdeputes.fr/jean-francois-lamour/json",
      },
      "2": Object {
        "date_naissance": "1954-10-31",
        "groupe_sigle": "SER",
        "id": 2,
        "id_an": "336439",
        "lieu_naissance": "Mauprevoir (Vienne)",
        "mandat_debut": "2012-06-20",
        "nb_mandats": 2,
        "nom": "Jean-Michel Clément",
        "nom_circo": "Vienne",
        "nom_de_famille": "Clément",
        "num_circo": 3,
        "num_deptmt": "86",
        "parti_ratt_financier": "Parti socialiste",
        "place_en_hemicycle": "263",
        "prenom": "Jean-Michel",
        "profession": "Avocat",
        "sexe": "H",
        "sites_web": Array [
          "https://twitter.com/jmclement86",
          "http://www.jeanmichelclement.com",
        ],
        "slug": "jean-michel-clement",
        "twitter": "jmclement86",
        "url_an": "http://www2.assemblee-nationale.fr/deputes/fiche/OMC_PA336439",
        "url_nosdeputes": "http://www.nosdeputes.fr/jean-michel-clement",
        "url_nosdeputes_api": "http://www.nosdeputes.fr/jean-michel-clement/json",
      },
    },
    "site": Object {
      "http://www.jeanmichelclement.com": Object {
        "site": "http://www.jeanmichelclement.com",
      },
      "http://www.jflamour.fr": Object {
        "site": "http://www.jflamour.fr",
      },
      "https://twitter.com/JF_Lamour": Object {
        "site": "https://twitter.com/JF_Lamour",
      },
      "https://twitter.com/jmclement86": Object {
        "site": "https://twitter.com/jmclement86",
      },
    },
  },
  "result": Array [
    1,
    2,
  ],
}
`;
